### DISCUSSION TITLE
New Contributor Help

### DISCUSSION THREAD

**Discussion by parthdpa** [pytest-dev/pytest#8524] [Q&A]
I am a current final year student and am looking to contribute to pytest as part of a final project. I am new to contributing to open source projects, so sorry for a lot simple questions and thank you for your time.

- For this project I am to make a contribution that changes the projects source code. 
- The contribution can be a feature request or a bug that is reasonable for a beginner to complete in 2 weeks and is documented as an issue or request. 
- The contribution cannot be trivial, so it should be relatively large so that I am spending at least 24 hours on the project that can include planning my approach and reflecting on my contribution
- It should include some QA on my part like testing. 
- I am also required to submit a PR for the issue/task I pick, but it does not necessarily have to be accepted. 

I am on a short timeline to submit a plan and task, so I appreciate any quick replies and insights, but it is fine if not.
Also, I am not looking for someone to plan and determine all these things for me, but am looking for insight and advice and for someone to point me in the right direction and issues to look at that seem reasonable.

I thank everyone for their time and input.

**Comment by RonnyPfannschmidt** [comment]
pinging @pytest-dev/core as i took note that the topic slipping me due to priorities

**Comment by pfctdayelise** [comment]
Hi parthdpa, have you used pytest before? What is your experience so far? Have you written for example your own fixtures or plugins? Have you encountered any difficulties using pytest yourself?

Have you looked at the issue tracker? Does anything particular pique your interest?

How confident are you in Python? If you browse the pytest source code does it seem very confusing, or are you able to start to make sense of it?

Plugins may be a good place to look as well as these are often widely used but tend to get less attention. eg. pytest-html.

**Reply by goodboy** [reply]
> Plugins may be a good place to look as well as these are often widely used but tend to get less attention.

I think this is good advice.

As a pre-note, `pytest` itself is built as set of plugins so it might worth understanding first how that sub-system works, play with some simpler plugins, then take a deeper dive into `pytest`'s core code base.

I know that's how I started and getting the plugin fundamentals clear in mind first helped a lot 🏄🏼
